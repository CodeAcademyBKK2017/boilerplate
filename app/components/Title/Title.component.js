import React, {Component} from 'react';
import titleStyle from './Title.style';
import {
  Text,
  TextInput,
  View
} from 'react-native';

export default class Content extends Component {
  render () {
    return (
      <View style={titleStyle.boxTitle}>
        <Text style={titleStyle.textTitle}>
          New Title
        </Text>
        <TextInput underlineColorAndroid='transparent' style={titleStyle.inputText}/>
      </View>
    );
  }
}
