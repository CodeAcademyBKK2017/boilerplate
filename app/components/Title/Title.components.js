import React, {Component} from 'react';
import styles from './Title.style';
import {
  Text,
  TextInput,
  View
} from 'react-native';

class Title extends Component {
  render () {
    return (
      <View style={styles.title}>
        <View style={styles.textRow}>
          <View style={styles.titleLeft}>
            <Text style={styles.titleText}>Note Title</Text>
          </View>
    
          <View >
            <Text style={styles.titleEn}>en</Text>
          </View>
        </View>
        
        <TextInput 
          style={styles.titleInput}
          placeholder='Tasks for today'
        />
       
      </View>
    );
  }
}

export default Title;