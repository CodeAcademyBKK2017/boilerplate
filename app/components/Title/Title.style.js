import {
  StyleSheet
} from 'react-native';
  
const titleStyles = StyleSheet.create({
  inputBox: {
    minWidth: '100%',
    height: 30,
    borderColor: 'gray',
    borderWidth: 1,
    marginTop: 20,
    backgroundColor: 'white'
  },
  fontTitle: {
    fontWeight: 'bold'
  }
});
  
export default titleStyles;