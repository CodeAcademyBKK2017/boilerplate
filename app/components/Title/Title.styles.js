import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    paddingBottom: 10
  },
  title: {
    fontSize: 13,
    color: '#000',
    fontWeight: 'bold'
  },
  textInput: {
    marginTop: 10,
    marginBottom: 10,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: '#c4c4c5',
    borderRadius: 3,
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 12,
    paddingRight: 12,
    backgroundColor: '#ffffff'
  }
});