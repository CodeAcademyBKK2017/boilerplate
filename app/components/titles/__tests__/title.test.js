import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

// Note: test renderer must be required after react-native.
import Title from '../title.component';
// import {shallow} from 'enzyme';

describe('Title', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Title />
    );
    expect(tree).toBeDefined();
  });
  it('Title: renders correctly', () => {
    const tree = renderer.create(<Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
