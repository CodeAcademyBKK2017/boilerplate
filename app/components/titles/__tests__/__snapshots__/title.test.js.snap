// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Title: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Note Title
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
      >
        EN
      </Text>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    style={
      Object {
        "backgroundColor": "#F5FCFF",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 30,
        "width": "100%",
      }
    }
  />
</View>
`;
